drivers/can/stm32_can.c

```diff
@@ -0,0 +1,765 @@
+
+#define SYS_LOG_LEVEL CONFIG_SYS_LOG_CAN_LEVEL
+#include <logging/sys_log.h>
+
+static const struct can_driver_api can_api_funcs = {
+	.configure = can_stm32_runtime_configure,
+	.send = can_stm32_send,
+	.attach_msgq = can_stm32_attach_msgq,
+	.attach_isr = can_stm32_attach_isr,
+	.detach = can_stm32_detach
+};
+
+static void signal_tx_complete(struct can_mailbox *mb)
+{
+	if (mb->tx_callback) {
+		mb->tx_callback(mb->error_flags);
+	} else  {
+		k_sem_give(&mb->tx_int_sem);
+	}
+}
+
+static inline void can_get_msg_fifo(CAN_FIFOMailBox_TypeDef *mbox,
+				    struct can_msg *msg)
+{
+	if (mbox->RIR & CAN_RI0R_IDE) {
+		msg->ext_id = mbox->RIR >> CAN_RI0R_EXID_Pos;
+		msg->id_type = CAN_EXTENDED_IDENTIFIER;
+	} else {
+		msg->std_id =  mbox->RIR >> CAN_RI0R_STID_Pos;
+		msg->id_type = CAN_STANDARD_IDENTIFIER;
+	}
+
+	msg->rtr = mbox->RIR & CAN_RI0R_RTR ? CAN_REMOTEREQUEST : CAN_DATAFRAME;
+	msg->dlc = mbox->RDTR & (CAN_RDT0R_DLC >> CAN_RDT0R_DLC_Pos);
+	msg->data_32[0] = mbox->RDLR;
+	msg->data_32[1] = mbox->RDHR;
+}
+
+static void can_isr(void *arg)
+{
+	struct device *dev;
+	struct can_stm32_data *data;
+	const struct can_stm32_config *cfg;
+	CAN_TypeDef *can;
+
+	__ASSERT_NO_MSG(arg != NULL);
+
+	dev = (struct device *)arg;
+	data = DEV_DATA(dev);
+	cfg = DEV_CFG(dev);
+	can = cfg->can;
+
+	__ASSERT_NO_MSG(data != NULL);
+
+	u32_t bus_off = can->ESR & CAN_ESR_BOFF;
+
+	if ((can->TSR & CAN_TSR_RQCP0) | bus_off) {
+		data->mb0.error_flags = can->TSR & CAN_TSR_TXOK0 ? CAN_TX_OK  :
+					can->TSR & CAN_TSR_TERR0 ? CAN_TX_ERR :
+					can->TSR & CAN_TSR_ALST0 ? CAN_TX_ARB_LOST :
+					bus_off ? CAN_TX_BUS_OFF :
+					CAN_TX_UNKNOWN;
+		/* clear the request. */
+		can->TSR |= CAN_TSR_RQCP0;
+		signal_tx_complete(&data->mb0);
+	}
+
+	if ((can->TSR & CAN_TSR_RQCP1) | bus_off) {
+		data->mb0.error_flags = can->TSR & CAN_TSR_TXOK1 ? CAN_TX_OK  :
+					can->TSR & CAN_TSR_TERR1 ? CAN_TX_ERR :
+					can->TSR & CAN_TSR_ALST1 ? CAN_TX_ARB_LOST :
+					bus_off ? CAN_TX_BUS_OFF :
+					CAN_TX_UNKNOWN;
+		/* clear the request. */
+		can->TSR |= CAN_TSR_RQCP1;
+		signal_tx_complete(&data->mb1);
+	}
+
+	if ((can->TSR & CAN_TSR_RQCP2) | bus_off) {
+		data->mb2.error_flags = can->TSR & CAN_TSR_TXOK2 ? CAN_TX_OK  :
+					can->TSR & CAN_TSR_TERR2 ? CAN_TX_ERR :
+					can->TSR & CAN_TSR_ALST2 ? CAN_TX_ARB_LOST :
+					bus_off ? CAN_TX_BUS_OFF :
+					CAN_TX_UNKNOWN;
+		/* clear the request. */
+		can->TSR |= CAN_TSR_RQCP2;
+		signal_tx_complete(&data->mb2);
+	}
+
+	if (can->TSR & CAN_TSR_TME_Msk) {
+		k_sem_give(&data->tx_int_sem);
+	}
+	while (can->RF0R & CAN_RF0R_FMP0_Msk) {
+		struct can_msg msg;
+		CAN_FIFOMailBox_TypeDef *mbox = &can->sFIFOMailBox[0];
+		int filter_match_index = ((mbox->RDTR & CAN_RDT0R_FMI_Msk)
+					  >> CAN_RDT0R_FMI_Pos);
+
+		if (filter_match_index >= CONFIG_CAN_MAX_FILTER) {
+			break;
+		}
+
+		SYS_LOG_DBG("Message on filter index %d", filter_match_index);
+		can_get_msg_fifo(mbox, &msg);
+		if (data->rx_response[filter_match_index]) {
```

Comment:
break a line before the if.
The only time you can pack the lines like that is when the if condition is testing previous line's assignment. Works the same for while/for 

Apply that everywhere.